src/jobs/SyncOktaUsersJob.php

```diff
@@ -76,16 +77,26 @@ public function getTitle()
      */
     public function process()
     {
-        // make an api call to get the list of users from okta
-        $users = Injector::inst()
-            ->get(OktaService::class)
-            ->getAllUsers(100, Config::inst()->get(SyncOktaUsersJob::class, 'statuses_to_sync'));
+        // For each of the user group filters make an api call to get the list of users from okta
+        $allGroupIDs = OktaUserGroupFilter::get();
+        $allUsers = [];
+        foreach ($allGroupIDs as $allGroupID) {
+            $users = Injector::inst()
+                ->get(OktaService::class)
```

Comment:
does this actually work?   presumably equivalent to what it's doing here and reads cleaner

$oktaService = Injector::singleton(OktaService::class) above the foreach block would be better

Reply:
Hmmm I gave this a go but it doesn't work.